src/Components/Blazor/Build/src/targets/Blazor.MonoRuntime.targets

```diff
@@ -128,25 +108,58 @@
+      The mechanism behind this (or whether it's correct) is a bit unclear so
+      https://github.com/dotnet/aspnetcore/issues/18951 tracks the need for follow-up.
+    -->
     <ItemGroup>
       <!--
-        ReferenceCopyLocalPaths includes all files that are part of the build out with CopyLocalLockFileAssemblies on.
-        Remove assemblies that are inputs to calculating the assembly closure. Instead use the resolved outputs, since it is the minimal set.
-       -->
-      <_BlazorCopyLocalPaths Include="@(ReferenceCopyLocalPaths)" />
-      <_BlazorCopyLocalPaths Remove="@(_BlazorManagedRuntimeAssemby)" />
+          ReferenceCopyLocalPaths includes all files that are part of the build out with CopyLocalLockFileAssemblies on.
+          Remove assemblies that are inputs to calculating the assembly closure. Instead use the resolved outputs, since it is the minimal set.
+         -->
+      <_BlazorCopyLocalPaths Include="@(ReferenceCopyLocalPaths)" Condition="'%(Extension)' == '.dll'" />
+      <_BlazorCopyLocalPaths Remove="@(_BlazorManagedRuntimeAssemby)" Condition="'%(Extension)' == '.dll'" />
 
       <BlazorOutputWithTargetPath Include="@(_BlazorCopyLocalPaths)">
         <BlazorRuntimeFile>true</BlazorRuntimeFile>
         <TargetOutputPath>$(BlazorRuntimeBinOutputPath)%(_BlazorCopyLocalPaths.DestinationSubDirectory)%(FileName)%(Extension)</TargetOutputPath>
         <RelativeOutputPath>%(_BlazorCopyLocalPaths.DestinationSubDirectory)%(FileName)%(Extension)</RelativeOutputPath>
       </BlazorOutputWithTargetPath>
+    </ItemGroup>
 
-      <BlazorOutputWithTargetPath Include="@(_BlazorResolvedAssembly)">
+    <ItemGroup>
+      <MonoWasmFile Include="$(DotNetWebAssemblyRuntimePath)*" />
+      <BlazorJSFile Include="$(BlazorJSPath)" />
+      <BlazorJSFile Include="$(BlazorJSMapPath)" Condition="Exists('$(BlazorJSMapPath)')" />
+
+      <BlazorOutputWithTargetPath Include="@(MonoWasmFile)">
+        <TargetOutputPath>$(BlazorRuntimeWasmOutputPath)%(FileName)%(Extension)</TargetOutputPath>
         <BlazorRuntimeFile>true</BlazorRuntimeFile>
-        <TargetOutputPath>$(BlazorRuntimeBinOutputPath)%(FileName)%(Extension)</TargetOutputPath>
-        <RelativeOutputPath>%(FileName)%(Extension)</RelativeOutputPath>
       </BlazorOutputWithTargetPath>
+      <BlazorOutputWithTargetPath Include="@(BlazorJSFile)">
+        <TargetOutputPath>$(BaseBlazorRuntimeOutputPath)%(FileName)%(Extension)</TargetOutputPath>
+      </BlazorOutputWithTargetPath>
+    </ItemGroup>
+
+    <ItemGroup Label="Static content supplied by NuGet packages">
+      <_BlazorPackageContentOutput Include="@(BlazorPackageContentFile)" Condition="%(SourcePackage) != ''">
+        <TargetOutputPath>$(BaseBlazorPackageContentOutputPath)%(SourcePackage)\%(RecursiveDir)\%(Filename)%(Extension)</TargetOutputPath>
+      </_BlazorPackageContentOutput>
+      <BlazorOutputWithTargetPath Include="@(_BlazorPackageContentOutput)" />
     </ItemGroup>
```

Comment:
How do we debug Release builds? Is that a thing we want to support? I can debug regular .net apps in Release configuration.

Reply:
Not sure how the question relates to this PR. Is it a general question about whether `pdb` files should be emitted and loaded in release builds? If so, a totally reasonable question but let's find a separate channel for discussing it, as this PR doesn't change anything about that.

Reply:
https://github.com/dotnet/aspnetcore/pull/18859/files#diff-c4620899f2dac172f7b494a8eb643b11R344-R355

Wouldn't we expose the pdbs unconditionally before?

Reply:
I don't really follow the link you've posted, since the code on those lines doesn't seem to say anything about whether pdbs are exposed.

**However** there was an issue with controlling whether PDBs are loaded: https://github.com/dotnet/aspnetcore/issues/18655. I've now amended `Blazor.MonoRuntime.targets` to fix this. Specifically, I changed how we filter out PDBs from the output so that it's controlled by the `BlazorEnableDebugging` in the same way whether linking is enabled or disabled.